packages/features/auth/signup/handlers/selfHostedHandler.ts

```diff
@@ -72,9 +74,26 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
         parent: true,
       },
     });
+
     if (team) {
+      const isInviteForOrganization = isOrganization({
+        team: {
+          metadata: team.metadata,
+        },
+      });
+      const isInviteForATeamInOrganization = team.parent;
+      const isCheckingUsernameInGlobalNamespace = !isInviteForOrganization && !isInviteForATeamInOrganization;
+
       const teamMetadata = teamMetadataSchema.parse(team?.metadata);
 
+      if (isCheckingUsernameInGlobalNamespace) {
+        const isUsernameAvailable = !(await isUsernameReservedDueToMigration(correctedUsername));
```

Comment:
Redirect is only applicable for global namespace and not inside an organization namespace.